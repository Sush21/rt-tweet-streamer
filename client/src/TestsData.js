export const tweet = {
  created_at: 'Thu Jan 28 01:30:09 +0000 2021',
  id_str: '1354602630899781634',
  full_text: '@VP: Get vaccinated. Save lives. https://t.co/5ZNXerlmIJ',
  favorite_count: 132354,
  description: '46th President of the United States, husband to @FLOTUS, proud dad & pop.',
  name: 'President Biden',
  profile_banner_url: 'https://pbs.twimg.com/profile_banners/1349149096909668363/1611162285',
  profile_image_url: 'http://pbs.twimg.com/profile_images/1349837426626330628/CRMNXzQJ_normal.jpg',
  screen_name: 'POTUS',
};

const tweet2 = {
  created_at: 'Thu Jan 28 19:50:01 +0000 2021',
  description: '46th President of the United States, husband to @FLOTUS, proud dad & pop. Tweets may be archived: https://t.co/IURuMIrzxb',
  favorite_count: 36703,
  full_text: 'Todayâ€™s actions are the first step in ensuring every American has access to quality, affordable health care. https://t.co/Csj6nQxc9m',
  id_str: '1354879423632334851',
  name: 'President Biden',
  profile_banner_url: 'https://pbs.twimg.com/profile_banners/1349149096909668363/1611162285',
  profile_image_url: 'http://pbs.twimg.com/profile_images/1349837426626330628/CRMNXzQJ_normal.jpg',
  screen_name: 'POTUS',
};

const tweet3 = {
  created_at: 'Thu Jan 28 01:30:09 +0000 2021',
  description: '46th President of the United States, husband to @FLOTUS, proud dad & pop. Tweets may be archived: https://t.co/IURuMIrzxb',
  favorite_count: 132756,
  full_text: 'One week down. https://t.co/yxrkWAXlkL',
  id_str: '1354602630899781634',
  name: 'President Biden',
  profile_banner_url: 'https://pbs.twimg.com/profile_banners/1349149096909668363/1611162285',
  profile_image_url: 'http://pbs.twimg.com/profile_images/1349837426626330628/CRMNXzQJ_normal.jpg',
  screen_name: 'POTUS',
};

const tweets = [];
for (let i = 0; i < 7; i += 1) tweets.push(tweet, tweet2, tweet3);
tweets.pop();
export default tweets;
