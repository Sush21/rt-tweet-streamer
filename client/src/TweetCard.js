import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { kFormatter, checkForLinks } from './UseHelper';

// Display tweet content
const TweetCard = ({ tweet }) => (
  <div className="ui feed">
    <div className="ui divider" />
    <div className="event">
      <div className="label">
        <img
          className="ui left floated mini circular image"
          src={tweet.profile_image_url}
          alt={tweet.name}
        />
      </div>
      <div className="content">
        <div className="summary">
          {tweet.name}
          <a
            href={`https://twitter.com/${tweet.screen_name}`}
            target="_blank"
            rel="noreferrer"
          >
            {' '}
            @
            {tweet.screen_name}
          </a>
          <div className="date">
            <a
              href={`https://twitter.com/${tweet.screen_name}/status/${tweet.id_str}`}
              target="_blank"
              rel="noreferrer"
            >
              {moment(tweet.created_at).startOf('day').fromNow()}
            </a>
          </div>
        </div>
        <div className="extra text">
          {checkForLinks(tweet.full_text)}
        </div>
        <div className="meta">
          <div className="like">
            <b>{kFormatter(tweet.favorite_count)}</b>
            {' '}
            Likes
          </div>
        </div>
      </div>
    </div>
  </div>
);

TweetCard.propTypes = {
  tweet: PropTypes.shape({
    profile_image_url: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    screen_name: PropTypes.string.isRequired,
    created_at: PropTypes.string.isRequired,
    full_text: PropTypes.string.isRequired,
    favorite_count: PropTypes.number.isRequired,
  }).isRequired,

};
export default TweetCard;
