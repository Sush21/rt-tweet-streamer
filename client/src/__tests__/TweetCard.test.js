import React from 'react';
import { shallow } from 'enzyme';
import TweetCard from '../TweetCard';
import { tweet } from '../TestsData';

describe('TweetCard', () => {
  it('should be defined', () => {
    expect(TweetCard).toBeDefined();
  });
  it('should render correctly', () => {
    const wrapper = shallow(
      <TweetCard tweet={tweet} />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('img').length).toEqual(1);
    expect(wrapper.find('img').props().src).toEqual(tweet.profile_image_url);
    expect(wrapper.find('div').first().hasClass('feed')).toEqual(true);
    expect(wrapper.find('a').length).toEqual(4);
    expect(wrapper.find('a').at(2).props().href).toEqual('https://twitter.com/VP');
    expect(wrapper.find('.extra').length).toEqual(1);
    expect(wrapper.find('b').text()).toEqual('132.4k');
  });
});
