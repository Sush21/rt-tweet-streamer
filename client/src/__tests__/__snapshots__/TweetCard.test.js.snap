// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetCard should render correctly 1`] = `ShallowWrapper {}`;
